src/tests/ffi-enc.cpp

```diff
@@ -735,6 +735,119 @@ TEST_F(rnp_tests, test_ffi_decrypt_pk_unlocked)
     rnp_ffi_destroy(ffi);
 }
 
+#if defined(ENABLE_CRYPTO_REFRESH)
+TEST_F(rnp_tests, test_ffi_decrypt_v6_pkesk_test_vector)
+{
+    rnp_ffi_t    ffi = NULL;
+    rnp_input_t  input = NULL;
+    rnp_output_t output = NULL;
+
+    assert_rnp_success(rnp_ffi_create(&ffi, "GPG", "GPG"));
+    assert_true(import_all_keys(ffi, "data/test_v6_valid_data/transferable_seckey_v6.asc"));
+
+    assert_rnp_success(rnp_input_from_path(&input, "data/test_v6_valid_data/v6pkesk.asc"));
+    assert_non_null(input);
+    assert_rnp_success(rnp_output_to_null(&output));
+    assert_rnp_success(rnp_decrypt(ffi, input, output));
+
+    // cleanup
+    rnp_input_destroy(input);
+    rnp_output_destroy(output);
+}
+
+TEST_F(rnp_tests, test_ffi_encrypt_pk_with_v6_key)
+{
+    rnp_ffi_t        ffi = NULL;
+    rnp_input_t      input = NULL;
+    rnp_output_t     output = NULL;
+    rnp_op_encrypt_t op = NULL;
+    const char *     plaintext = "data1";
+
+    // setup FFI
+    assert_rnp_success(rnp_ffi_create(&ffi, "GPG", "GPG"));
+
+    assert_true(import_all_keys(ffi, "data/test_v6_valid_data/transferable_seckey_v6.asc"));
+
+    // RNP_LOG_HEX("key id", ffi->secring->keys.front().keyid().data(),
+    // ffi->secring->keys.front().keyid().size()); RNP_LOG_HEX("key id",
+    // ffi->secring->keys.back().keyid().data(), ffi->secring->keys.back().keyid().size());
+
+    std::vector<std::string> ciphers = {"AES128", "AES192", "AES256"};
+    std::vector<std::string> aead_modes = {"None", "EAX", "OCB"};
+    std::vector<bool>        enable_pkeskv6_modes = {true, false};
+
+    for (auto enable_pkeskv6 : enable_pkeskv6_modes)
```

Comment:
Let's leave it as it is, but such approach could lead to the problem where it is impossible to tell which exact combination of enable_pkeskv6/aead_modes/ciphers lead to failure. We usually duplicate code in tests, adding some helper functions to do not make tests code too bloated.